### PR TITLE
HPCC-27666 Expose deployment name via ECL (Std.System.Job.deployment())

### PR DESCRIPTION
**Pull request by jakesmith**
For use in ECL to tell which environment the query is running in.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: roxie/ccd/ccdactivities.cpp [hunk 1]
```diff
@@ -588,6 +588,7 @@ class CRoxieAgentActivity : implements CInterfaceOf<IRoxieAgentActivity>, implem
     // The below are called on Roxie server and passed in context
     virtual unsigned getPriority() const { throwUnexpected(); }
     virtual char *getPlatform() { throwUnexpected(); }
+    virtual char *getDeploymentName() { throwUnexpected(); }
```
Comment (ghalliday) [inline roxie/ccd/ccdactivities.cpp:591]: Unlikely to be called, but I suspect cleaner to implement as a call like getDaliServers().

File: roxie/ccd/ccdcontext.cpp [hunk 2]
```diff
@@ -3754,6 +3755,14 @@ class CRoxieServerContext : public CRoxieContextBase, implements IRoxieServerCon
             return strdup("roxie");
 #endif
     }
+    virtual char *getDeploymentName()
```
Comment (ghalliday) [inline roxie/ccd/ccdcontext.cpp:3758]: @richardkchapman would this be better implemented in CRoxieContextBase?

File: helm/hpcc/templates/_helpers.tpl [hunk 3]
```diff
@@ -190,6 +190,7 @@ Pass in root as .
 {{- $certificates := (.Values.certificates | default dict) -}}
 {{- $issuers := ($certificates.issuers | default dict) -}}
 {{- $security := .Values.security | default dict -}}
+deployment: {{ (include "hpcc.fullname" (dict "root" $)) }}
```
Comment (ghalliday) [inline helm/hpcc/templates/_helpers.tpl:193]: Can this also be configured in baremetal?  Where would the option go?

Comment (ghalliday) [inline helm/hpcc/templates/_helpers.tpl:193]: Can this also be configured in baremetal?  Where would the option go?

**Review by ghalliday — COMMENTED** [review]
Code looks correct.  One question for Richard about the best placement n roxie, and one question about bare-metal

File: roxie/ccd/ccdactivities.cpp (hunk 1 above)
Comment (jakesmith) [reply roxie/ccd/ccdactivities.cpp:591]: agreed, I'll change.

Comment (jakesmith) [reply roxie/ccd/ccdactivities.cpp]: @ghalliday - have pushed that 1 line change (and squashed it in):
```
-    virtual char *getDeploymentName() { throwUnexpected(); }
+    virtual char *getDeploymentName() { return getDaliServers(); }
```

**Review by ghalliday — COMMENTED** [review]
Approved, with a question for Richard

**Review by ghalliday — APPROVED** [review]

**Comment by ghalliday** [comment]
@richardkchapman waiting for a response to a question.  @jakesmith probably needs to target 8.8.x now.

**Comment by ghalliday** [comment]
@richardkchapman merging ready for 8.8.0, but it would still be worth you checking the code in roxie.

**Pull request merged**